### PR TITLE
quiz api to fetch status from other packages

### PR DESCRIPTION
**Pull request by ice-cronus**

### CODE REVIEW

File: kyc/quiz/api/quiz.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		return res, nil
+	}
+	i := len(params)
+	for _, userID := range userIDs {
+		params = append(params, userID)
+		placeholders = append(placeholders, fmt.Sprintf("$%v", i+1))
+		i++
+	}
+
+	sql := fmt.Sprintf(`SELECT u.id,
```
Comment (ice-ares) [inline kyc/quiz/api/quiz.go]: Make sure the sql is performant. It needs to be fast, adapt it if it's not

Comment (ice-cronus) [reply kyc/quiz/api/quiz.go]: with 1000 random users (as I remember batch size in freezer) it is ~150 ms for a cold request, ~30-40 with a warmed

**Review by ice-myles — DISMISSED** [review]

File: kyc/quiz/quiz.go [hunk 2]
```diff
@@ -80,18 +81,34 @@ func NewRepository(ctx context.Context, userRepo UserRepository) Repository {
...
-func newRepositoryImpl(ctx context.Context, userRepo UserRepository) *repositoryImpl {
+func NewReadRepository(ctx context.Context) ReadRepository {
 	db := storage.MustConnect(ctx, ddl, applicationYamlKey)
 
-	return &repositoryImpl{
+	return &readRepository{
 		DB:       db,
 		Shutdown: db.Close,
-		Users:    userRepo,
 		config:   mustLoadConfig(),
```
Comment (ice-ares) [inline kyc/quiz/quiz.go]: Build the cfg differently for read and write, so that u required only what u need for each

File: kyc/quiz/quiz.go [hunk 3]
```diff
@@ -256,18 +273,48 @@ func (r *repositoryImpl) CheckQuizStatus(ctx context.Context, userID UserID) (*Q
...
+		*r.config.MaxResetCount,
+		r.config.MaxAttemptsAllowed,
+	}, make([]string, 0, len(userIDs)), make(map[string]*QuizStatus, len(userIDs))
+	if len(userIDs) == 0 {
+		return res, nil
+	}
+	i := len(params)
+	for _, userID := range userIDs {
+		params = append(params, userID)
+		placeholders = append(placeholders, fmt.Sprintf("$%v", i+1))
```
Comment (ice-ares) [inline kyc/quiz/quiz.go]: No, bad ideea, use []string as sql parameter and id = any($xxx) in sql

File: kyc/quiz/quiz.go [hunk 4]
```diff
@@ -277,30 +324,33 @@ func (r *repositoryImpl) getQuizStatus(ctx context.Context, userID UserID) (*Qui
...
+		*QuizStatus
+		UserID string `db:"id"`
+	}](
 		ctx,
 		r.DB,
 		sql,
-		userID,
-		r.config.globalStartDate.Time,
-		r.config.AvailabilityWindowSeconds,
-		*r.config.MaxResetCount,
```
Comment (ice-ares) [inline kyc/quiz/quiz.go:292]: Rollback that part after u add the new []string parameter

**Review by ice-ares — CHANGES_REQUESTED** [review]

File: kyc/quiz/quiz.go [hunk 5]
```diff
@@ -256,13 +277,32 @@ func (r *repositoryImpl) CheckQuizStatus(ctx context.Context, userID UserID) (*Q
 	return status, err
 }
 
-func (r *repositoryImpl) getQuizStatus(ctx context.Context, userID UserID) (*QuizStatus, error) { //nolint:funlen //.
-	// $1: user_id.
-	// $2: global start date.
```
Comment (ice-ares) [inline kyc/quiz/quiz.go:261]: Keep the same order

File: kyc/quiz/quiz.go [hunk 6]
```diff
@@ -256,13 +277,32 @@ func (r *repositoryImpl) CheckQuizStatus(ctx context.Context, userID UserID) (*Q
...
+	quizStatus, found := statuses[userID]
+	if found && quizStatus != nil {
+		quizStatus.KYCQuizAvailable = quizStatus.KYCQuizAvailable && (r.isKYCEnabled(ctx) || r.isKYCStepForced(userID))
+	}
+
+	return quizStatus, errors.Wrapf(err, "failed to getQuizStatus for userID:%v", userID)
+}
+
+func (r *readRepository) GetQuizStatus(ctx context.Context, userIDs ...string) (map[string]*QuizStatus, error) { //nolint:funlen //.
+	// $1: global start date.
```
Comment (ice-ares) [inline kyc/quiz/quiz.go]: Most likely those comments are wrong

**Review by ice-ares — CHANGES_REQUESTED** [review]

**Pull request merged**